crates/ursa-consensus/src/config.rs

```diff
@@ -1,73 +1,155 @@
 // Copyright 2022-2023 Fleek Network
 // SPDX-License-Identifier: Apache-2.0, MIT
 
-use std::{path::PathBuf, sync::Arc};
+#![allow(clippy::mutable_key_type)]
 
-use fastcrypto::{bls12381::min_sig::BLS12381KeyPair, ed25519::Ed25519KeyPair};
 use multiaddr::Multiaddr;
-use mysten_metrics::RegistryService;
-use narwhal_config::{Parameters, WorkerId};
-use narwhal_crypto::NetworkKeyPair as NarwhalNetworkKeyPair;
+use narwhal_config::{
+    Authority, Committee, Parameters, Stake, WorkerCache, WorkerId, WorkerIndex, WorkerInfo,
+};
+use narwhal_crypto::{NetworkPublicKey, PublicKey};
 use serde::{Deserialize, Serialize};
-use tokio::sync::OnceCell;
+use std::time::Duration;
+use std::{collections::BTreeMap, path::PathBuf};
 
-pub type KeyPair = Ed25519KeyPair;
-pub type AuthorityKeyPair = BLS12381KeyPair;
-
-#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
-pub struct ValidatorKeyPair {
-    #[serde(skip)]
-    keypair: OnceCell<Arc<AuthorityKeyPair>>,
+#[derive(Debug, Clone, Deserialize, Serialize)]
+// When deserializing the config file, use the default from the Default instance
+// to fill any missing field.
+#[serde(default)]
+pub struct ConsensusConfig {
+    /// The address in which the primary will listen for incoming requests on. This MUST
+    /// be a UDP address.
+    pub address: Multiaddr,
```

Comment:
Should we provide individual field defaults for future upgradability? Mirroring what the other crates provide, for example:
https://github.com/fleek-network/ursa/blob/main/crates/ursa-rpc-service/src/config.rs#L4-L17

Reply:
This can be done separately too, otherwise LGTM

Reply:
we do have defaults, i'm just using the  `#[serde(default)]` over the entire struct which fallbacks to `Default` impl on the type, i think this is cleaner.